PR: Rewrote configuration to emphasis singular `App` class usage

bbrennanbasis opened the pull request:
This PR closes #231.

hyyan commented on docs/getting-started/configuration.md:
```diff
@@ -38,29 +38,43 @@ When using the installation plugin, the tags within the `<configuration>` tag ca
 - **`<debug>`** The debug tag can be set to true or false, and will determine whether or not the browser's console displays error messages thrown by your program. 
 
-### Running a specific application
+### Running a specific program
```
@MatthewHawkins I believe this section is misplaced and doesn't belong in the configuration section. Additionally, routing is optional anyway. what do you think ?

hyyan commented:
@bbrennanbasis It looks good. Please review all the Vale issues on this page and ensure they are fixed. Thank you!

bbrennanbasis commented:
@hyyan I removed the specific step-by-step instructions for routing and mentioned the other way of running a specific app (using the class that extends `App`).

hyyan commented:
@MatthewHawkins, please review this and check if it aligns with the plans for documenting the new configuration approach. It might be best to hold off until you've completed that.

MatthewHawkins commented:
@hyyan I read over the changes here and it looks better to me. I'm not opposed to merging before the new WAR configuration is documented, as I believe enforcing this behavior with the routing changes implemented is sufficient, but if you'd prefer to wait until that's completed, then no problem.

hyyan commented:
@bbrennanbasis requested minor formatting changes
@MatthewHawkins merge when ready please

bbrennanbasis commented:
@hyyan 
- Changed application to app
- Removed misleading wording talking about with and without the plugin
- Rewrote Default class section and changed the header name
